mountpoint-s3/tests/fuse_tests/fork_test.rs

```diff
@@ -380,6 +384,130 @@ fn mount_scoped_credentials() -> Result<(), Box<dyn std::error::Error>> {
+    let (bucket, prefix) = get_test_bucket_and_prefix("mount_and_check_log");
+
+    let mount_point = assert_fs::TempDir::new().expect("can not create a mount dir");
+    let child = mount_with_sse(&bucket, mount_point.path(), &prefix, &sse_key, Some(credentials));
+    erroneous_write(mount_point.path());
+    unmount_and_check_log(child, mount_point.path(), &expected_log_line);
+}
+
+#[cfg(all(feature = "sse_kms", not(feature = "s3express_tests")))]
+#[test]
+fn read_with_no_permissions_for_a_key() {
+    // create file
+    let (bucket, prefix) = get_test_bucket_and_prefix("mount_and_check_log");
+    let mut test_client = crate::common::fuse::s3_session::create_test_client(&get_test_region(), &bucket, &prefix);
+    let encrypted_object = "encrypted_with_kms";
+    let unencrypted_object = "unencrypted_with_s3_keys";
+    let data = vec![0xaa; 32];
+    let sse_key = get_test_kms_key_id();
+    test_client
+        .put_object_params(
+            encrypted_object,
+            &data,
+            mountpoint_s3_client::types::PutObjectParams::new()
+                .server_side_encryption(Some("aws:kms:dsse".to_string()))
+                .ssekms_key_id(Some(sse_key.clone())),
+        )
+        .expect("failed to create an object in S3");
+    test_client
+        .put_object(unencrypted_object, &data)
+        .expect("failed to create an object in S3");
+
+    // try to read it
+    let log_line_pattern = format!("^.*WARN.*{encrypted_object}.*read failed: get request failed: get object request failed: Client error: Forbidden: User: .* is not authorized to perform: kms:Decrypt on resource: {sse_key} because no session policy allows the kms:Decrypt action.*$");
+    let expected_log_line = regex::Regex::new(&log_line_pattern).unwrap();
+    let policy_with_no_kms_perms = r#"{"Statement": [
+        {"Effect": "Allow", "Action": ["s3:*"], "Resource": "*"}
+    ]}"#;
+    let credentials = tokio_block_on(get_scoped_down_credentials(policy_with_no_kms_perms));
+    let mount_point = assert_fs::TempDir::new().expect("can not create a mount dir");
+    let child = mount_with_sse(&bucket, mount_point.path(), &prefix, &sse_key, Some(credentials));
+    {
+        let encrypted_object = mount_point.join(encrypted_object);
+        let mut data = Vec::new();
+        let mut f = fs::File::open(encrypted_object).expect("can not open file for read");
+        let read_result = f.read_to_end(&mut data);
+        assert!(
+            read_result.is_err(),
+            "should not be able to read a kms-encrypted file without kms permissions"
+        );
+
+        let unencrypted_object = mount_point.join(unencrypted_object);
+        let mut f = fs::File::open(unencrypted_object).expect("can not open file for read");
+        let read_result = f.read_to_end(&mut data);
+        assert!(
+            read_result.is_ok(),
+            "should not able to read a default-encrypted file after the first read failure"
+        );
+    }
```

Comment:
why is this in a scoped block? not against it, just the reason isn't clear

I guess to close the files? let's add a comment to the scope

Reply:
exactly for that, will add a comment